DemoQA/Tests/elements_webtables.js

```diff
@@ -20,72 +21,91 @@ test(`Check page validity`, async t => {
 
 test(`Add a user`, async t => {
     await webTablesPage.navigateToWebTablesPage("Elements", "Web Tables");
-    await registrationFormPage.addNewUser();
+    await webTablesPage.addNewUser(newUserDetails);
 
-    const newUserRow = new UserInfoRow(3); //to modify it to not be hardocaded on a line (index needed) + check defaut reviewer pentru LAszlo , command lines in testcafe to video and screenshots si reports/ config file cu setat de paths, size window, time in teste 
+    const newUserRow = new UserInfoRow(3); //modify it to not be hardocaded on a line (index needed) + check defaut reviewer pentru LAszlo , command lines in testcafe to video and screenshots si reports/ config file cu setat de paths, size window, time in teste 
     const userDetails = await newUserRow.getUserDetails();
 
     await t
-        .expect(userDetails.firstName).eql(myUserDetails.firstName)
-        .expect(userDetails.lastName).eql(myUserDetails.lastName)
-        .expect(userDetails.age).eql(myUserDetails.age)
-        .expect(userDetails.emailAddress).eql(myUserDetails.email)
-        .expect(userDetails.salary).eql(myUserDetails.salary)
-        .expect(userDetails.department).eql(myUserDetails.department);
+        .expect(userDetails.firstName).eql(newUserDetails.firstName)
+        .expect(userDetails.lastName).eql(newUserDetails.lastName)
+        .expect(userDetails.age).eql(newUserDetails.age)
+        .expect(userDetails.emailAddress).eql(newUserDetails.email)
+        .expect(userDetails.salary).eql(newUserDetails.salary)
+        .expect(userDetails.department).eql(newUserDetails.department);
 });
 
 
 test(`Delete a user`, async t => {
     await webTablesPage.navigateToWebTablesPage("Elements", "Web Tables");
       
-    const currentUsersList = await webTablesPage.usersTableList();
+    const currentUsersList = await webTablesPage.filterValidUserDetails();
 
-    await registrationFormPage.addNewUser();
+    await webTablesPage.addNewUser(newUserDetails);
  
     const latestUser = new UserInfoRow(3);
 
     await latestUser.deleteUser();
 
-    const latestUsersList = await webTablesPage.usersTableList();
+    const latestUsersList = await webTablesPage.filterValidUserDetails();
 
     await t
-        .expect(webTablesPage.usersTable.innerText).notContains('Mia')
+        .expect(webTablesPage.usersTable.innerText).notContains('Felix')
         .expect(currentUsersList).eql(latestUsersList);
 });
 
 
 test(`Edit a new user`, async t => {
     await webTablesPage.navigateToWebTablesPage("Elements", "Web Tables");
-    await registrationFormPage.addNewUser();
+    await webTablesPage.addNewUser(myUserDetails);
 
     const newUser = new UserInfoRow(3);
 
     await newUser.editUser();
-    await registrationFormPage.clearFormFields(); 
-    await registrationFormPage.fillRegistrationForm(newUserDetails);
-    await registrationFormPage.submitForm();
+    await registrationFormPage.fillRegistrationForm(myUserDetails);
```

Comment:
This test will never fail, because you create the user with myUserDetails and you edit it to the user myUserDetails, so none of the values actually change.